kytos/core/controller.py

```diff
@@ -627,9 +627,9 @@ def set_switch_options(self, dpid):
         if not switch:
             return
 
-        vlan_pool = {}
+        vlan_pool = self.options.vlan_pool
         try:
-            vlan_pool = json.loads(self.options.vlan_pool)
+            # vlan_pool = json.loads(self.options.vlan_pool)
```

Comment:
Remove this line, don't leave it as a comment.

Reply:
Something strange is happening in tox... simply removing the comment is causing test_switch to fail. Any idea?

Reply:
The test `test_switch_vlan_pool_default` is testing against a string because that was the default value, as it was later used in a call to `json.loads`. Now that you changed the behavior, and `json.loads` is called when loading the config, the value stored is a `dict` and not a `string` anymore. The test needs to be refactored to reflect that.

Reply:
I see.. so that's what I thought towards the end of my last commits last night. I wasn't sure if I was allowed to modify the unit tests. I'll refractor them accordingly and see if we can get the tests to pass for this PR. Thank you for your help, @ajoaoff !

Reply:
Fixed! Thank you for your help Antonio!